fix: Update condition_index field of unmatched feature flag

Josh-IE opened the pull request:
Set the `/decide?v=4` endpoint to return null on the `condition_index` field, when a feature flag condition is not met.

## Problem

<!-- Who are we building for, what are their needs, why is this important? -->

<!-- Does this fix an issue? Uncomment the line below with the issue ID to automatically close it when merged -->
Closes #30569

## Changes

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

## Does this work well for both Cloud and self-hosted?

<!-- Yes / no / it doesn't have an impact. -->

## How did you test this code?

Tested locally

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

Josh-IE committed:
Return null for feature flag condition_index field

Set the decide api endpoint to return null on the `condition_index` field, when a feature flag condition is not met.

Josh-IE committed:
Merge branch 'master' into fix/feature-flag-match

Josh-IE committed:
Merge branch 'master' into fix/feature-flag-match

Josh-IE committed:
Merge branch 'master' into fix/feature-flag-match

Josh-IE committed:
Update test

Josh-IE committed:
Update feature flag evaluation test

Josh-IE committed:
Merge branch 'master' into fix/feature-flag-match

haacked commented:
Thanks for the contribution! Before we merge this, we need to make sure all the clients will handle a null value for that field correctly. I'll do that before merging this.

haacked committed:
Set the `condition_index` to `None` when not matching a condition

The same goes for super conditions since it's not
one of the flag conditions that matched.

haacked committed:
Merge branch 'master' into fix/feature-flag-match

haacked commented:
I pushed relevant changes to the Rust flag service. @dmarticus can you review the rust changes?

dmarticus reviewed (approved):
sorry for the delay on this!
